src/excmds.ts

```diff
@@ -3296,6 +3296,31 @@ export async function qall() {
     return Promise.all(windows.map(window => browser.windows.remove(window.id)))
 }
 
+//#background
+export async function sessionsave(...name: string[]){
+    let namestr = name.join(" ")
+    let tabs = await browser.tabs.query({})
+    let sessions = {}
+    for (let tab of tabs){
+        sessions[tab.windowId] = sessions[tab.windowId] || []
+        sessions[tab.windowId].push(tab.url)
+    }
+    let sessionstr = JSON.stringify(sessions)
+    return config.set("sessions", namestr, sessionstr)
+}
+
+//#background
```

Comment:
Background/async? What if there are *lots* of tabs or there is heavy CPU pressure and Firefox is responding slow? Will the user get any feedback while session is being saved? When the session is saved?